src/pymovements/stimulus/text.py

```diff
@@ -0,0 +1,94 @@
+# Copyright (c) 2023-2024 The pymovements Project Authors
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""Module for the TextDataFrame."""
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Any
+
+import polars as pl
+
+from pymovements.stimulus.stimulus import Stimulus
+
+
+class TextStimulus(Stimulus):
+    """TextStimulus class
+    """
+
+    def __init__(
+            self,
+            aois: pl.DataFrame,
+            character_column: str,
+            pixel_x_column: str,
+            pixel_y_column: str,
+            width_column: str,
+            height_column: str,
+            page_column: str,
+    ) -> None:
+
+        self.frame = aois.select(
+            pl.col(character_column),
+            pl.col(pixel_x_column),
+            pl.col(pixel_y_column),
+            pl.col(width_column),
+            pl.col(height_column),
+            pl.col(page_column),
+        )
+
+
+def from_file(
+        aoi_path: str | Path,
+        *,
+        character_column: str,
+        pixel_x_column: str,
+        pixel_y_column: str,
+        width_column: str,
+        height_column: str,
+        page_column: str,
+        custom_read_kwargs: dict[str, Any] | None = None,
+) -> TextStimulus:
+    """ Load text stimulus from file"""
+    if isinstance(aoi_path, str):
+        aoi_path = Path(aoi_path)
```

Comment:
polars supports str as path type, so I don't think this if is necessary

Reply:
we need this to check for suffix -- I can also change it to split at dot and check for the suffix differently?